tests/i18n/tests.py

```diff
@@ -259,8 +259,14 @@ def test_maclines(self):
             self.assertEqual('Catalan Win\nEOF\n', gettext('Win\r\nEOF\r\n'))
 
     def test_to_locale(self):
+        """
```

Comment:
I'd use inline comments rather than a docstring. I also added a failing case:
```
tests = (
    ('en-us', 'en_US'),
    # A langage with > 2 characters after the dash.
    ('sr-lat', 'sr_Lat'),
    # Languages with a private-use subtag (x-informal).
    ('nl-nl-x-informal', 'nl_NL-x-informal'),
    ('sr-lat-x-informal', 'sr_LAT-x-informal'),
)
for lang, output in tests:
    with self.subTest(lang=lang):
        self.assertEqual(to_locale(lang), output)
```